core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -360,5 +360,7 @@ private TableProperties() {}
   public static final String UPSERT_ENABLED = "write.upsert.enabled";
   public static final boolean UPSERT_ENABLED_DEFAULT = false;
 
-  public static final String HMS_TABLE_OWNER = "hms_table_owner";
+  public static final String HMS_TABLE_OWNER = "hive.metastore.table.owner";
```

Comment:
@haizhou-zhao thanks for updating the property names, but now looking more closely at how these are used, I don't think they should live in the class.  

Two of the properties are database properties (not table properties) and they are all hive specific.  We're trying to keep the properties here isolated to general properties for tables (I know there are a few spark ones, but they should probably move as well).

I think it makes sense just to put these as properties in the HiveCatalog class or create a HiveCatalogProperties to encapsulate these behaviors. 

Other than that, this looks good.

Reply:
thanks @danielcweeks , I'll put them in HiveCatalog class.

Reply:
If this goes in before cutting 1.1, we're fine.  If it goes in after, then we just need to mark that deprecated for removal in 1.2.

Reply:
Just to be on the safe side I opened a one-liner PR to rename the HMS_TABLE_OWNER that we can squeeze into 1.1.0 easier and once that's in we don't have to break backward compatibility.
https://github.com/apache/iceberg/pull/6154